File: mobile/lib/models/post.dart
import 'package:mobile/models/user.dart'; // 작성자 정보를 담기 위해 User 모델을 가져옵니다.

class Post {
  final String id;
  final String title;
  final String? content;
  final String userId;
  final DateTime createdAt;
  final User? author; // 작성자 정보 (nullable)

  Post({
    required this.id,
    required this.title,
    this.content,
    required this.userId,
    required this.createdAt,
    this.author,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      title: json['title'],
      content: json['content'],
      userId: json['user_id'],
      createdAt: DateTime.parse(json['created_at']),
      // author 필드가 null이 아닐 경우에만 User.fromJson을 호출합니다.
      author: json['author'] != null ? User.fromJson(json['author']) : null,
    );
  }
}

File: backend/app/schemas/post.py
from pydantic import BaseModel
from datetime import datetime
import uuid
from typing import List, Optional

from .user import User  # 댓글 작성자 정보를 표시하기 위해 User 스키마를 가져옵니다.

# ====== Comment Schemas ======
class CommentBase(BaseModel):
    content: str

class CommentCreate(CommentBase):
    pass

class Comment(CommentBase):
    id: uuid.UUID
    user_id: uuid.UUID
    post_id: uuid.UUID
    created_at: datetime
    author: Optional[User] = None # 댓글에 작성자 정보 포함

    class Config:
        from_attributes = True

# ====== Post Schemas ======
class PostBase(BaseModel):
    title: str
    content: Optional[str] = None

class PostCreate(PostBase):
    pass

class Post(PostBase):
    id: uuid.UUID
    user_id: uuid.UUID
    created_at: datetime
    author: Optional[User] = None # 게시글에 작성자 정보 포함
    comments: List[Comment] = [] # 게시글 상세 조회 시 댓글 목록 포함

    class Config:
        from_attributes = True